src/layered_config_tree/main.py

```diff
@@ -357,6 +357,22 @@ def get(self, key: str, default_value: NodeValue = None) -> LayeredConfigTree |
         default = LayeredConfigTree() if default_value is None else default_value
         return self[key] if key in self._children else default
 
+    def get_tree(self, key: str, default_value: NodeValue = None) -> LayeredConfigTree:
```

Comment:
Why did you give this a `default_value` argument with a default value of `NodeValue`? This will always return a `LayeredConfigTree` (or error), so if we hit a default value, we are guaranteed to error.